templates/ovncontroller/bin/start-vswitchd.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/sh
+#
+# Copyright 2024 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+source $(dirname $0)/functions
+wait_for_ovsdb_server
+
+# The order - first wait for db server, then set -ex - is important. Otherwise,
+# wait_for_ovsdb_server interrim check would make the script exit.
+set -ex
+
+# Configure encap IP.
+OVNEncapIP=$(ip -o addr show dev {{ .OVNEncapNIC }} scope global | awk '{print $4}' | cut -d/ -f1)
+ovs-vsctl --no-wait set open . external-ids:ovn-encap-ip=${OVNEncapIP}
+
+# Before starting vswitchd, block it from flushing existing datapath flows.
+ovs-vsctl --no-wait set open_vswitch . other_config:flow-restore-wait=true
+
+# It's safe to start vswitchd now. Do it.
+# --detach to allow the execution to continue to restoring the flows.
+/usr/sbin/ovs-vswitchd --pidfile --mlockall --detach
+
+# Restore saved flows.
+if [ -f $FLOWS_RESTORE_SCRIPT ]; then
+    # It's unsafe to leave these files in place if they fail once. Make sure we
+    # remove them if the eval fails.
+    trap cleanup_flows_backup EXIT
+    for bridge_flows in `find $ovs_dir -name *flows.dump`; do
+        echo Bridge to print: $bridge_flows
+        cat $bridge_flows
```

Comment:
is this dump just done for tests or we need these in stdout normally too?

Reply:
I think folks were suggesting that it's helpful to have this in logs. (Not that I agree, just answering that it's probably intentional.) At the very least, if this `cat` stays, a comment should be added explaining that it's not an oversight but an intentional change.

I think there may be some risk in dumping what effective is network configuration for a particular chassis in logs. I wonder if anything sensitive is there. I understand there are no passwords, but there may be domain names configured by users, DHCP options, perhaps something else? AFAIU the logs are then fed into logging system. Are we comfortable dumping flow tables there?

Reply:
If we use the approach proposed of adding a new bridge, this flow dump won't be needed.